api_spec.yml

```diff
@@ -0,0 +1,967 @@
+            type: string
+            description: URL pointing to the previous exercise in the collection.
+          submissions:
+            type: string
+            description: URL pointing to a collection of all submissions for the exercise.
+      title:
+        type: string
+        description: A title of the exercise.
+      readme:
+        type: string
+        description: Contents of readme file for the exercise.
+      track:
+        $ref: '#/definitions/Track'
+      tests:
+        type: array
+        description: Test file for the exercise.
+        items:
+          properties:
+            filename:
+              type: string
+              description: A name of the test file.
+            contents:
+              type: string
+              description: A contents of the test file.
+
+  # Exercise collection model
+  ExercisesCollection:
+    properties:
+      _type:
+        type: string
+        description: A type of the collection - 'exercisesCollection'.
+      _links:
+        type: object
+        properties:
+          self:
+            type: string
+            description: URL pointing to the current page of the collection.
+          first:
+            type: string
+            description: URL pointing to the first page of the collection.
+          last:
+            type: string
+            description: URL pointing to the last page of the collection.
+          next:
+            type: string
+            description: URL pointing to the next page of the collection.
+          previous:
+            type: string
+            description: URL pointing to the previous page of the collection.
+      _total:
+        type: integer
+        format: int32
+        description: Total number of resources in the collection.
+      _page:
+        type: integer
+        format: int32
+        description: Number of current page.
+      _perPage:
+        type: integer
+        format: int32
+        description: Number of resources per page.
+      data:
+        type: array
+        description: Exercises in the collection.
+        items:
+          $ref: '#/definitions/Exercise'
+
+  # Submission model
+  Submission:
+    properties:
+      _type:
+        type: string
+        description: A type of the resource - 'submission'.
+      _links:
+        type: object
+        properties:
+          self:
+            type: string
+            description: URL pointing to the resource.
+          first:
+            type: string
+            description: URL pointing to the first submission in the collection.
+          last:
+            type: string
+            description: URL pointing to the last submission in the collection.
+          next:
+            type: string
+            description: URL pointing to the next submission in the collection.
+          previous:
+            type: string
+            description: URL pointing to the previous submission in the collection.
+          update:
```

Comment:
Submissions are not updated per se (the code remains untouched). It might get more views, likes, or comments, but that seems like a separate thing. At the moment the state can change ("close/reopen"), though this concept might go away later.